### PR TITLE
[bugfix] Set token metadata

### PR DESCRIPTION
**Pull request by jiyoung-an**
## Summary
- confirm the `mint` transaction is finalized and set the token metadata.
- poll the transaction state every 10 seconds, with a timeout of 3 minutes.

### CODE REVIEW

File: controllers/api.js [hunk 1]
```diff
@@ -93,9 +94,23 @@ class APIController {
 
       const { tx_hash } = await object.mint(checksum, tokenId.toString());
 
+      const startTime = Date.now();
+      const timeout = 3 * 60 * 1000; // 3min
```
Comment (platfowner) [inline controllers/api.js:98]: how about defining a constant variable?

File: controllers/api.js [hunk 2]
```diff
@@ -93,9 +94,23 @@ class APIController {
...
+      while (true) {
+        const result = await ain.getTransactionByHash(tx_hash);
+        console.log(result.state);
+        if (result.state === 'FINALIZED') {
+          break;
+        }
+        await sleep(10 * 1000); // 10sec
```
Comment (platfowner) [inline controllers/api.js:105]: ditto?

File: controllers/message.js [hunk 3]
```diff
@@ -107,7 +107,7 @@ class MessageController {
       } else {
         console.log(run.status);
       }
-      await new Promise((resolve) => setTimeout(resolve, 100)); // 0.1sec
+      await sleep(100); // 0.1sec
```
Comment (platfowner) [inline controllers/message.js:110]: ditto?

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks!

File: controllers/api.js (hunk 1 above)
Comment (jiyoung-an) [reply controllers/api.js]: done!

File: controllers/api.js (hunk 2 above)
Comment (jiyoung-an) [reply controllers/api.js]: done!

File: controllers/message.js (hunk 3 above)
Comment (jiyoung-an) [reply controllers/message.js]: done!

**Comment by jiyoung-an** [comment]
Thank you for quick review @platfowner! now merging...

**Pull request merged**